src/dsp_tools/utils/xml_upload_stash.py

```diff
@@ -0,0 +1,264 @@
+    verbose: bool,
+    con: Connection,
+) -> dict[XMLResource, dict[XMLProperty, dict[str, KnoraStandoffXml]]]:
+    """
+    This function uploads a single xml link property, which was previously stashed.
+
+    Args:
+        link_prop_in_triplestore: the link property from the triplestore
+        res_iri: the iri of the resource
+        stashed_resource: the stashed resource
+        resource_in_triplestore: the resource retrieved from the triplestore
+        link_prop: the name of the link property
+        hash_to_value: the has value of the xml text
+        id2iri_mapping: the dictionary with the internal ids and the new IRIs
+        nonapplied_xml_texts: the dictionary with the stashes
+        verbose: what is printed out
+        con: the connection to the triplestore
+
+    Returns:
+        The stash dictionary with the newly uploaded resource removed.
+        If the upload was not sucessfull it returns the dictionary as it was before.
+    """
+    xmltext_in_triplestore = link_prop_in_triplestore.get("knora-api:textValueAsXml")
+    if not xmltext_in_triplestore:
+        # no action necessary: this property will remain in nonapplied_xml_texts,
+        # which will be handled by the caller
+        return nonapplied_xml_texts
+
+    # strip all xml tags from the old xmltext, so that the pure text itself remains
+    text_hash_value = _get_text_hash_value(xmltext_in_triplestore)
+
+    # if the pure text is a hash, the replacement must be made
+    # this hash originates from _stash_circular_references(), and identifies the XML texts
+    if text_hash_value not in hash_to_value:
+        # no action necessary: this property will remain in nonapplied_xml_texts,
+        # which will be handled by the caller
+        return nonapplied_xml_texts
+
+    new_xmltext = _replace_internal_ids_with_iris(
+        pure_text=text_hash_value, id2iri_mapping=id2iri_mapping, hash_to_value=hash_to_value
+    )
+
+    # prepare API call
+    jsondata = _create_XMLResource_json_object_to_update(
+        res_iri=res_iri,
+        resource_in_triplestore=resource_in_triplestore,
+        stashed_resource=stashed_resource,
+        link_prop_in_triplestore=link_prop_in_triplestore,
+        new_xmltext=new_xmltext,
+        link_prop_name=link_prop.name,
+    )
+
+    # execute API call
+    try:
+        try_network_action(con.put, route="/v2/values", jsondata=jsondata)
+    except BaseError as err:
+        _log_unable_to_upload_xml_resource(
+            received_error=err, stashed_resource=stashed_resource, all_link_props=link_prop
+        )
+        return nonapplied_xml_texts
+    if verbose:
+        print(f'  Successfully uploaded xml text of "{link_prop.name}"\n')
+        logger.info(f'  Successfully uploaded xml text of "{link_prop.name}"\n')
+    nonapplied_xml_texts[stashed_resource][link_prop].pop(text_hash_value)
+    return nonapplied_xml_texts
+
+
+def iterate_over_all_link_props_of_single_resource(
```

Comment:
This name misses the core point of what the function does. The function **uploads** all link props of a resource. If it iterates over it doesn't matter at all.